### ISSUE TITLE
Media storage config for S3 backend in existingSecret

### ISSUE THREAD

**Issue by dimitrigraf** [bootc/netbox-chart#56]
The chart allows to have almost all of the credentials in an existing secret and therefore not to have them in the `values.yaml`.

The only credentials that can not yet be extracted like that are the credentials (`storageConfig`) for the media storage backend.

Enabling that would be much appreciated. Thanks in advance!

Cheers.

**Comment by bootc** [comment]
The trouble is this requires domain-specific knowledge for the storage provider(s) you want to use.

Really this chart needs a way to add arbitrary additional configuration settings from other ConfigMaps and Secrets or similar, but that's not straightforward. I'll put some thought into it.

**Comment by bootc** [comment]
My plan for this is to let people supply arbitrary ConfigMaps and Secrets, and any `*.yaml` keys within those secrets will be treated as configuration, and imported into NetBox's configuration. I'd like to do similar for `*.py` but that may need to wait for a future update as I don't want to delay releasing version 4.0 of the chart.

**Comment by bootc** [comment]
I've just released version [4.0.1](https://github.com/bootc/netbox-chart/releases/tag/4.0.1) of this chart which resolves this issue.

**Comment by dimitrigraf** [comment]
Hi there @bootc 

I'm not sure if you'll see this if the issue is closed but I'll try anyway. 

So, I was trying out the `extraConfig` field today because I wanted to extract the S3 config to an existing secret like discussed above. But, at some point I realized that this is not going to work because the mechanism creates files with the filename being the key and the value being the file content. 

I just tried and get a botocore exception saying that the credentials couldn't be found. The container does have the files though:

```
bash-5.1$ ls /run/config/extra/0/
AWS_ACCESS_KEY_ID      AWS_SECRET_ACCESS_KEY
```

Did I misunderstand something? I guess so...
